### DISCUSSION TITLE
Is there a way of carrying out preprocessing using quantile transformation in river?

### DISCUSSION THREAD

**Discussion by medha-chippa** [online-ml/river#692] [Q&A] [answered]
Can preprocessing using quantile transformation be performed using river? In other words, can data within a stream be transformed using quantile transformation?

**Comment by MaxHalford** [comment, marked as answer]
Nope, not at the moment. We have a `stats.Quantile` class to compute online quantiles, so this should be totally doable. Pull requests are welcome!

Although, have you tried `preprocessing.StandardScaler`? I'd be curious to see a case where quantile transformation is really necessary.

**Reply by medha-chippa** [reply]
No, I have not tried preprocessing.StandardScaler. Quantile transformations are especially useful when the train and test data come from different distributions/domains but have the same predictors.

**Reply by MaxHalford** [reply]
What does the "same predictors" mean?

I see the problem you're trying to solve, but I don't see how quantile transformation provides an advantage over standard scaling, which is way more common. Also, this train/test is not apparent in online learning. Test data becomes training data once it's labelled, and the training and test phases are interleaved.

**Reply by medha-chippa** [reply]
Same predictors here refers to the same features/columns in the dataset. Yes, I understand that there is no train/test data in online learning. That's exactly why I wish to perform preprocessing on the stream object (one data point) that arrives with no distinction between train/test. I have tried doing the same using the adaptive standard scaler module, the code to which has been included in one of my other questions.
